src/lib/User/model.ts

```diff
@@ -0,0 +1,68 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+export enum UserRole {
+   Admin = "admin",
+   Mentor = "mentor",
+   Mentee = "mentee",
+}
+
+export enum MeetingPreference {
+   Remote = "Remote",
+   InPerson = "InPerson",
+}
+
+export interface User extends Document {
+   name: string;
+   email: string;
+   password: string;
+   userIntro: string;
+   userRole: string;
+   userSkills: [];
+   yearsOfProfessionalExperience: number;
+   timeZoneUTCOffset: number;
+   availability: string;
+   meetingPreference: string;
+}
+
+const UserSchema: Schema<User> = new Schema({
+   name: {
+      type: String,
+      required: [true, "Username is required"],
+   },
+   email: {
+      type: String,
+      required: [true, "Email is required"],
+      unique: true,
+   },
+   password: {
+      type: String,
+      required: true,
+   },
+   userIntro: {
+      type: String,
+      required: true,
+   },
+   userRole: {
+      type: String,
+      required: true,
+   },
+   userSkills: {
+      type: [String],
+   },
+   yearsOfProfessionalExperience: {
+      type: Number,
+   },
+   timeZoneUTCOffset: {
+      type: Number,
```

Comment:
We want to ensure this value can only be within the range of `-12 - 12`

Reply:
validation added